### DISCUSSION TITLE
We should use codecov or sonarcloud for showing code coverage project badge.

### DISCUSSION THREAD

**Discussion by anantk24** [microsoft/gctoolkit#52] [General]
As above tools already has functionality of showing badges of code coverage , code smells, project maintenance.

Please check the sample links for reference

- [CodeCov](https://about.codecov.io/product/feature/badges/)
- SonarCloud
   - [![Coverage](https://sonarcloud.io/api/project_badges/measure?project=AzureAD_microsoft-identity-web&metric=coverage)](https://sonarcloud.io/dashboard?id=AzureAD_microsoft-identity-web)
   - [![Maintainability Rating](https://sonarcloud.io/api/project_badges/measure?project=AzureAD_microsoft-identity-web&metric=sqale_rating)](https://sonarcloud.io/dashboard?id=AzureAD_microsoft-identity-web)
> We can avoid creating project badges manually and pushing that if there are any changes, above badges are updated automatically when code is pushed.

Ref #50

**Comment by ghost** [comment]
I'm all for better use of these tools. That said, I think the tools can promote behaviours that take us to extremes. For example, I'm not a fan of 100% test coverage as I find this is a deterrent to refactoring. As crazy as that may sound, I am a fan of more black box testing that focuses on APIs and results produced by using those APIs. There are also important but components that should be (mostly) stable that should likely be very heavy guarded with tests. In the case of GCToolkit, this applies to the parsers and I think this focus is demonstrated in the test coverage numbers. That said, I'm all for PRs that adds more/better test coverage.
